cpu/sam0_common/sam0-eth/eth-netdev.c

```diff
@@ -0,0 +1,164 @@
+int sam0_eth_receive_blocking(char *data, unsigned max_len);
+void sam0_eth_set_mac(const eui48_t *mac);
+void sam0_eth_get_mac(char *out);
+
+sam0_eth_netdev_t _sam0_eth_dev;
+
+static int _sam0_eth_init(netdev_t *netdev)
+{
+    sam0_eth_init();
+    eui48_t hwaddr;
+    netdev_eui48_get(netdev, &hwaddr);
+    sam0_eth_set_mac(&hwaddr);
+    return 0;
+}
+
+static void _sam0_eth_isr(netdev_t *netdev)
+{
+    netdev->event_callback(netdev, NETDEV_EVENT_RX_COMPLETE);
+    return;
+}
+
+static int _sam0_eth_recv(netdev_t *netdev, void *buf, size_t len, void *info)
+{
+    (void)info;
+    (void)netdev;
+    unsigned ret = sam0_eth_receive_blocking((char *)buf, len);
+
+    return ret;
+}
+
+static int _sam0_eth_send(netdev_t *netdev, const iolist_t *iolist)
+{
+    int ret;
+
+    netdev->event_callback(netdev, NETDEV_EVENT_TX_STARTED);
+    ret = sam0_eth_send(iolist);
+    if (ret == -EOVERFLOW) {
+        /* TODO: use a specific netdev callback here ? */
+        return -EOVERFLOW;
+    }
+    netdev->event_callback(netdev, NETDEV_EVENT_TX_COMPLETE);
+    return ret;
+}
+
+static int _sam0_eth_get(netdev_t *netdev, netopt_t opt, void *val, size_t max_len)
+{
+    int res = -1;
+
+    switch (opt) {
+        case NETOPT_ADDRESS:
+            assert(max_len >= ETHERNET_ADDR_LEN);
+            sam0_eth_get_mac((char *)val);
+            res = ETHERNET_ADDR_LEN;
+            break;
+        default:
+            res = netdev_eth_get(netdev, opt, val, max_len);
+            break;
+    }
+
+    return res;
+}
+
+static int _sam0_eth_set(netdev_t *netdev, netopt_t opt, const void *val, size_t max_len)
+{
+    int res = -1;
+
+    switch (opt) {
+        case NETOPT_ADDRESS:
+            assert(max_len >= ETHERNET_ADDR_LEN);
+            sam0_eth_set_mac((eui48_t *)val);
+            res = ETHERNET_ADDR_LEN;
+            break;
+        default:
+            res = netdev_eth_set(netdev, opt, val, max_len);
+            break;
+    }
+
+    return res;
+}
+
+static const netdev_driver_t _sam0_eth_driver =
+{
+    .send = _sam0_eth_send,
+    .recv = _sam0_eth_recv,
+    .init = _sam0_eth_init,
+    .isr  = _sam0_eth_isr,
+    .get  = _sam0_eth_get,
+    .set  = _sam0_eth_set,
+};
+
+void sam0_eth_setup(netdev_t* netdev)
+{
+
+    LOG_INFO("[sam0_eth]: initializing SAM0 Ethernet MAC (GMAC) device\n");
+
+    _sam0_eth_dev.netdev = netdev;
+    /* set the netdev driver */
+    netdev->driver = &_sam0_eth_driver;
+    /* Register SAM0 Ethernet to netdev */
+    netdev_register(netdev, NETDEV_SAM0_ETH, 0);
+}
+
+/* TODO: rework the whole isr management... */
+void isr_gmac(void)
+{
+    uint32_t isr;
+    uint32_t tsr;
+    uint32_t rsr;
+
+    isr = GMAC->ISR.reg;
+    tsr = GMAC->TSR.reg;
+    rsr = GMAC->RSR.reg;
+
+    if (rsr == GMAC_RSR_REC) {
+        netdev_trigger_event_isr(_sam0_eth_dev.netdev);
+    }
+
+    GMAC->TSR.reg = tsr;
+    GMAC->RSR.reg = rsr;
+    GMAC->ISR.reg = isr;
+    cortexm_isr_end();
```

Comment:
```suggestion

    cortexm_isr_end();
```

already started nitpicking whitespaces, so I might as well :wink: